### PR TITLE
Agnt 1398 App ID

### PR DESCRIPTION
**Pull request by vlukman**
Enhancement for https://crittercism.atlassian.net/browse/AGNT-1398

WORK DONE
- Added a PhoneGap method Crittercism.init(appID)
- Removed required IOS_APP_ID and ANDROID_APP_ID when installing PhoneGap plugin

TEST
- Installed plugin with the following command: cordova -d plugin add https://github.com/crittercism/PhoneGap#AGNT-1398-appID
- Called Crittercism.init(appID) on javascript

TEST RESULT
- App runs normally on iOS and Android without crash. Crittercism SDK was initialized normally.

@dshirley Please review
@sforen Please be very strict with my Java code

### CODE REVIEW

File: platforms/android/com/crittercism/plugin/CDVCrittercism.java [hunk 1]
```diff
@@ -57,16 +57,22 @@ public void initialize(CordovaInterface cordova, CordovaWebView webView) {
         Context context = this.cordova.getActivity();
         packageName = context.getPackageName();
         Resources resources = context.getResources();
-        String crAppID = context.getString(resources.getIdentifier(
-                APPLICATION_ID, STRING, packageName));
-        Crittercism.initialize(context, crAppID);
+        int identifier = resources.getIdentifier(APPLICATION_ID, STRING, packageName);
```
Comment (dshirley) [inline platforms/android/com/crittercism/plugin/CDVCrittercism.java]: For customers that were previously on the old system that switch to the new system, I assume their app ID will end up hanging around in their resources.  Would this result in the app getting initialized twice?  The native code protects against this, but they may get a warning message.

File: platforms/android/com/crittercism/plugin/CDVCrittercism.java [hunk 2]
```diff
@@ -98,6 +104,14 @@ public boolean execute(String action, final JSONArray args,
         }
     }
 
+    private boolean executeInit(String action, final JSONArray args,
+                               CallbackContext callbackContext) throws JSONException {
+        final String appID = args.getString(0);
```
Comment (dshirley) [inline platforms/android/com/crittercism/plugin/CDVCrittercism.java]: We should probably do some validation here and check if "args" has length >= 1.
A customer could potentially leave out the parameters on the javascript side.

**Comment by dshirley** [comment]
Looks good.  Couple minor comments.

File: platforms/android/com/crittercism/plugin/CDVCrittercism.java (hunk 1 above)
Comment (sforen) [inline platforms/android/com/crittercism/plugin/CDVCrittercism.java]: More than double-initialization, I wonder if this "old initialization" runs before the customer calls the new Crittercism.init(). So if they already have some app ID in their resources, will they be able to change it with "new initialization"?

Comment (vlukman) [inline platforms/android/com/crittercism/plugin/CDVCrittercism.java]: Yes, this code will cause double initialization. Fixing it won't be difficult, I am planning to prioritize the app ID on the xml/plist file.

Some interesting fact: to update PhoneGap plugins, each of the plugin needs to be removed and reinstalled. When a user removes or reinstalls Crittercism plugin, the app ID on the xml/plist file is also gone. Technically this code is not necessary. Thoughts?

File: platforms/android/com/crittercism/plugin/CDVCrittercism.java (hunk 2 above)
Comment (vlukman) [inline platforms/android/com/crittercism/plugin/CDVCrittercism.java]: I tested calling Crittercism.init with an empty argument on an Android device. Turns out PhoneGap (or javascript?) passed in 'null' string if user passes no arguments. We don't need a check here.

**Pull request merged**